bigquery/cloud-client/snippets.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env python
+
+# Copyright 2016 Google Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Samples that demonstrate basic operations in the BigQuery API.
+
+For more information, see the README.md under /bigquery.
+
+Example invocation:
+    $ python snippets.py list-datasets
+
+The dataset and table should already exist.
+"""
+
+import argparse
+
+from gcloud import bigquery
+
+
+def list_projects():
+    raise NotImplementedError(
+        'https://github.com/GoogleCloudPlatform/gcloud-python/issues/2143')
+
+
+def list_datasets(project=None):
+    """Lists all datasets in a given project.
+
+    If no project is specified, then the currently active project is used
+    """
+    bigquery_client = bigquery.Client(project=project)
+
+    datasets = []
+    page_token = None
+
+    while True:
+        results, page_token = bigquery_client.list_datasets(
+            page_token=page_token)
+        datasets.extend(results)
+
+        if not page_token:
+            break
+
+    for dataset in datasets:
+        print(dataset.name)
+
+
+def list_tables(dataset_name, project=None):
+    """Lists all of the tables in a given dataset.
+
+    If no project is specified, then the currently active project is used.
+    """
+    bigquery_client = bigquery.Client(project=project)
+    dataset = bigquery_client.dataset(dataset_name)
+
+    if not dataset.exists():
+        print('Dataset {} does not exist.'.format(dataset_name))
+
+    tables = []
+    page_token = None
+
+    while True:
+        results, page_token = dataset.list_tables(page_token=page_token)
+        tables.extend(results)
+
+        if not page_token:
+            break
+
+    for table in tables:
+        print(table.name)
+
+
+def list_rows(dataset_name, table_name, project=None):
+    """Prints rows in the given table.
+
+    Will print 25 rows at most for brevity as tables can contain large amounts
+    of rows.
+
+    If no project is specified, then the currently active project is used.
+    """
+    bigquery_client = bigquery.Client(project=project)
+    dataset = bigquery_client.dataset(dataset_name)
+    table = dataset.table(table_name)
+
+    if not table.exists():
+        print('Table {}:{} does not exist.'.format(dataset_name, table_name))
+
+    # Reload the table so that the schema is available.
+    table.reload()
+
+    rows = []
+    page_token = None
+
+    while len(rows) < 25:
```

Comment:
It's one extra line, generally considered a good practice to avoid magic numbers, and you use the magic number twice so you avoid the risk of changing it once place and not the other. So I still vote for the change but merge away if you're not convinced.